src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+)
+    return LineSearchesStepsize{
+        typeof(ls),typeof(is),typeof(retraction_method),typeof(vector_transport_method)
+    }(
+        ls, is, retraction_method, vector_transport_method
+    )
+end
+
+function (cs::StepsizeStorage{<:LineSearchesStepsize})(
+    mp::AbstractManoptProblem,
+    s::AbstractManoptSolverState,
+    i::Int,
+    η=-get_gradient(s);
+    fp=get_cost(mp, get_iterate(s)),
+    kwargs...,
+)
+    css = cs.stepsize
+    M = get_manifold(mp)
+    p = get_iterate(s)
+    X = get_gradient(s)
+    p_tmp = copy(M, p)
+    X_tmp = copy(M, p, η)
+    Y_tmp = copy(M, p, η)
+    f = get_cost_function(get_objective(mp))
+    dphi_0 = real(inner(M, p, X, η))
+    #println("dphi_0 = ", dphi_0)
+
+    # guess initial alpha
+    get_initial_alpha(M, css.initial_step, cs, p, η, fp, dphi_0)
+    α0 = cs.alpha
+
+    cs.last_f_p = fp
+
+    # perform actual linesearch
+
+    function ϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        return f(M, p_tmp)
+    end
+    function dϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        return real(inner(M, p_tmp, Y_tmp, Y_tmp))
+    end
+    function ϕdϕ(α)
+        # TODO: optimize?
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        phi = f(M, p_tmp)
+        dphi = real(inner(M, p_tmp, Y_tmp, Y_tmp))
+        return (phi, dphi)
+    end
+
+    try
+        α, fp = css.linesearch(ϕ, dϕ, ϕdϕ, α0, fp, dphi_0)
+        return α
+    catch ex
+        if isa(ex, LineSearches.LineSearchException)
+            # maybe indicate failure?
+            return zero(dphi_0)
+        else
+            rethrow(ex)
+        end
+    end
+end
+
+function show(io::IO, cs::LineSearchesStepsize)
+    return print(
+        io,
+        "LineSearchesStepsize($(cs.linesearch); retraction_method=$(cs.retraction_method), vector_transport_method=$(cs.vector_transport_method))",
+    )
+end
+
+"""
+    nanmax(x, y)
+
+Maximum of arguments, ignoring those that are NaN.
+"""
+function nanmax(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return max(x, y)
+end
+function nanmax(x, y, z)
+    return nanmax(nanmax(x, y), z)
+end
+
+"""
+    nanmin(x, y)
+
+Minimum of arguments, ignoring those that are NaN.
+"""
+function nanmin(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return min(x, y)
+end
+
+# code below was adapted from https://github.com/JuliaNLSolvers/LineSearches.jl/blob/master/src/initialguess.jl
+
+function get_initial_alpha(
+    M::AbstractManifold, is::InitialStatic{T}, state::StepsizeStorage, p, η, phi_0, dphi_0
+) where {T}
+    PT = promote_type(T, real(number_eltype(η)))
+    if is.scaled == true && (ns = real(norm(M, p, η))) > convert(PT, 0)
```

Comment:
So this case never happened, so why do we have it?

Reply:
That's copied from LineSearches.jl: https://github.com/JuliaNLSolvers/LineSearches.jl/blob/ded667a80f47886c77d67e8890f6adb127679ab4/src/initialguess.jl#L17 . I haven't checked in what cases it is useful.

Reply:
Do we need to copy it or is there – in the idea of a wrapper that we follow here – a wrapper also possible?

Reply:
It's not an exact copy. In this case I've replaced a standard `norm` with tangent vector norm, so a wrapper would have to translate that part as well. The code is very short so I think it's easier to copy it and change a little instead of making an elaborate wrapper.

Reply:
Ok, and we even might have to keep the name then, hm.